client/mm/libxc/binance.go

```diff
@@ -1636,49 +1769,69 @@ func (bnc *binance) connectToMarketDataStream(ctx context.Context, baseID, quote
 	}
 	book := newBinanceOrderBook(baseCfg.conversionFactor, quoteCfg.conversionFactor, mktID, getSnapshot, bnc.log)
 	bnc.books[mktID] = book
+	marketStreamsURL := bnc.marketStreamsURL()
 	bnc.booksMtx.Unlock()
 
-	// Create initial connection to the market data stream
-	conn, cm, err := newConnection()
+	// Need to send key but not signature
+	connectEventFunc := func(cs comms.ConnectionStatus) {
+		if cs != comms.Disconnected && cs != comms.Connected {
+			return
+		}
+
+		// If disconnected, set all books to unsynced so bots
+		// will not place new orders.
+		connected := cs == comms.Connected
+
+		bnc.booksMtx.RLock()
+		defer bnc.booksMtx.RUnlock()
+
+		for _, b := range bnc.books {
+			select {
+			case b.connectedChan <- connected:
+			default: // don't block
+			}
+		}
+	}
+
+	reconnectInterval := time.Hour * 12
+	conn, err := comms.NewWsConn(marketStreamsURL, &comms.WsCfg{
+		// Binance Docs: The websocket server will send a ping frame every 3
+		// minutes. If the websocket server does not receive a pong frame
+		// back from the connection within a 10 minute period, the connection
+		// will be disconnected. Unsolicited pong frames are allowed.
+		PingWait:     time.Minute * 4,
+		EchoPingData: true,
+		ReconnectSync: func() {
+			bnc.log.Debugf("Binance reconnected")
```

Comment:
OK not to trigger a full reconnect here, but maybe a call to `checkSubs` would be prudent.